crypto/txsigner_eip155.go

```diff
@@ -0,0 +1,95 @@
+package crypto
+
+import (
+	"crypto/ecdsa"
+	"math/big"
+	"math/bits"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type EIP155Signer struct {
+	chainID uint64
```

Comment:
Maybe `chainID` should be `int64` because that is the type used in configuration? Also, there a lot conversions later on when doing math operations

Reply:
I just moved an existing code to a separate file. I would love to change it but within a separate PR because this one is only about adding a new signer.